Call volume plugin directly

StrongMonkey opened the pull request.

Commit:
add version ability

Commit:
add test for new fields

ibuildthecloud committed:
Merge pull request #57 from StrongMonkey/solve-ping

add timeout for goroutine and http client

ibuildthecloud committed:
Merge pull request #59 from cjellick/host-label-env-var-fix

Fix host labels from env var

ibuildthecloud committed:
Merge pull request #53 from StrongMonkey/get-version

add version ability

Commit:
fix telemetry panic

ibuildthecloud committed:
Merge pull request #60 from StrongMonkey/tel-panic

fix telemetry panic

ibuildthecloud committed:
Merge pull request #56 from StrongMonkey/new-fields

Add test for new fields

Commit:
change test case

ibuildthecloud committed:
Merge pull request #61 from StrongMonkey/change-stupid-memory-fields

change test case

Commit:
update dependency

Commit:
add windows support

Commit:
add fields before set dns

Commit:
add timeout to all container list

ibuildthecloud committed:
Merge pull request #63 from StrongMonkey/fix-dns

add fields before set dns

ibuildthecloud committed:
Merge pull request #64 from StrongMonkey/add-timeout-to-all-docker-ps

add timeout to all container list

Commit:
fix dns problem

Commit:
enforce timeout to docker client except for storage one

Commit:
add backup method for fqdn

ibuildthecloud committed:
Handle volume errors on start

ibuildthecloud committed:
Merge pull request #65 from StrongMonkey/fix-dns-again

fix dns problem

Commit:
update dependency

ibuildthecloud committed:
Merge pull request #68 from ibuildthecloud/fixes

Handle volume errors on start

ibuildthecloud committed:
Merge pull request #67 from StrongMonkey/fqdn-coreos

add backup method for fqdn

ibuildthecloud committed:
Merge pull request #66 from StrongMonkey/timeout-again

enforce timeout to docker client except for storage one

Commit:
bump to 1.7.3

Commit:
clean log

Commit:
seperate test cases

ibuildthecloud committed:
Merge pull request #69 from StrongMonkey/go1.7

bump to 1.7.3

ibuildthecloud committed:
Merge pull request #71 from StrongMonkey/seperate-test

seperate test cases

Commit:
Merge pull request #70 from StrongMonkey/clean-log

clean log

Commit:
update vendor

Commit:
update engine-api to docker/docker/client

Commit:
Merge pull request #62 from StrongMonkey/windows-demo

Windows support

ibuildthecloud committed:
Merge pull request #72 from StrongMonkey/new-client

New client

ibuildthecloud committed:
Fixes for new networking

ibuildthecloud committed:
Remove test_instance_activate_links_no_service

ibuildthecloud committed:
Merge pull request #73 from ibuildthecloud/networking

Fixes for new networking

ibuildthecloud committed:
Delete unneeded code in new networking design

ibuildthecloud committed:
Add CNI support

ibuildthecloud committed:
Merge pull request #75 from ibuildthecloud/new-net

Delete unneeded code in new networking design

ibuildthecloud committed:
Assign CNI network name to label

Commit:
skip image pull variants

Commit:
Remove rancher/agent from ping test

ibuildthecloud committed:
Merge pull request #76 from ibuildthecloud/network2

Assign CNI network name to label

Commit:
add support for user namespace

Commit:
Merge pull request #77 from cjellick/skip-pull-variants

Fix or skip slow tests

Commit:
remove parse logic in agent

Commit:
Merge pull request #80 from StrongMonkey/image-fix

remove parse logic in agent

Commit:
Merge pull request #78 from StrongMonkey/user-namespace

add support for user namespace

Commit:
update vendor

Commit:
update getIP method

Commit:
delete testPrivateImage

Commit:
Merge pull request #82 from StrongMonkey/delete-test

delete testPrivateImage

Commit:
Test fixes

Commit:
Merge pull request #83 from cjellick/test-fixes

Test fixes

ibuildthecloud committed:
Fix wrong entrypoint insertion

ibuildthecloud committed:
Merge pull request #84 from ibuildthecloud/fix-entry

Fix wrong entrypoint insertion

ibuildthecloud committed:
Merge pull request #81 from StrongMonkey/new-ip

New getIP

ibuildthecloud committed:
When running net:none still return exposed ports

ibuildthecloud committed:
Merge pull request #85 from ibuildthecloud/fix-ports

When running net:none still return exposed ports

ibuildthecloud committed:
Get pid from docker

ibuildthecloud committed:
Merge pull request #86 from ibuildthecloud/fix

Get pid from docker

Commit:
image noop fix

ibuildthecloud committed:
Fix memory leaks

Both http.Client and NsHandle usage was causing FDs to leak.
Additionally the error handling for getIP was cleaned up

ibuildthecloud committed:
Merge pull request #89 from ibuildthecloud/fixes2

Fix memory leaks

Commit:
don't return error if noop

Commit:
short life container ip fix

ibuildthecloud committed:
Merge pull request #88 from StrongMonkey/fix

image noop fix

ibuildthecloud committed:
Merge pull request #90 from StrongMonkey/noop-fix

don't return error if noop

ibuildthecloud committed:
Support docker 1.10 and 1.11

ibuildthecloud committed:
Merge pull request #92 from ibuildthecloud/support-10-11

Support docker 1.10 and 1.11

ibuildthecloud committed:
Merge pull request #91 from StrongMonkey/fix1

short life container ip fix

ibuildthecloud committed:
Delete unneeded code

Commit:
Remove volume driver

Commit:
fix

Commit:
Merge pull request #95 from StrongMonkey/fix2

fix

Commit:
Merge pull request #94 from cjellick/vol-driver-fix

Remove volume driver

Commit:
Merge pull request #93 from ibuildthecloud/unneeded

Delete unneeded code

Commit:
new golint style

Commit:
add support for windows ci

dx9 committed:
Bring -static back

ibuildthecloud committed:
Merge pull request #96 from StrongMonkey/new-golint

new golint style

ibuildthecloud committed:
Revert "fix"

This reverts commit ff0f4ed1f5f437870df582b9133c76506605a49b.

ibuildthecloud committed:
Fix timeout on IP

ibuildthecloud committed:
Merge pull request #98 from dx9/static-link

Bring -static back

ibuildthecloud committed:
Merge pull request #100 from ibuildthecloud/fix2

Fix2

Commit:
delete unneeded code

Commit:
change uuid to container name

ibuildthecloud committed:
Merge pull request #102 from StrongMonkey/container-network

change uuid to container name

Commit:
Merge pull request #101 from StrongMonkey/delete

delete unneeded code

Commit:
update vendor

Commit:
update .gitignore and scripts

Commit:
migrating host-api

Commit:
windows support

Commit:
Merge pull request #97 from StrongMonkey/windows-ci

add support for windows ci

Commit:
Merge pull request #87 from StrongMonkey/add-host-api

Migrating host-api and add windows supports

Commit:
reflect change

Commit:
update vendor

Commit:
Merge pull request #104 from StrongMonkey/master

rancher/rancher#6957

Commit:
fix race condition

Commit:
don't use timeout for dockerclient when pulling images

Commit:
don't panic just pass as python-agent did

Commit:
use both cases for proxy envs

Commit:
env fix

Commit:
Do build during package if binary doesnt exist

ibuildthecloud committed:
Merge pull request #109 from StrongMonkey/envfix

env fix

ibuildthecloud committed:
Merge pull request #108 from StrongMonkey/proxy

use both cases for proxy envs

ibuildthecloud committed:
Merge pull request #107 from StrongMonkey/rancher6910

don't panic just pass as python-agent did

ibuildthecloud committed:
Merge pull request #106 from StrongMonkey/imagefix

don't use timeout for dockerclient when pulling images

ibuildthecloud committed:
Merge pull request #105 from StrongMonkey/racefix

fix race condition

ibuildthecloud committed:
Support non-/var/lib/docker root

ibuildthecloud committed:
Merge pull request #111 from ibuildthecloud/non-standard-root

Support non-/var/lib/docker root

Commit:
don't use hard code for image authConfig

ibuildthecloud committed:
Add service DNS first

ibuildthecloud committed:
Merge pull request #112 from StrongMonkey/auth-fix

don't use hard code for image authConfig

ibuildthecloud committed:
Merge pull request #113 from ibuildthecloud/dns

Add service DNS first

Commit:
set proxy only for system container and override it

Commit:
Merge pull request #110 from cjellick/fix-package-script

Do build during package if binary doesnt exist

Commit:
populate io.rancher.scheduler.ips label in each host

Commit:
Merge pull request #114 from StrongMonkey/proxy-fix

set proxy only for system container and override it

Commit:
add panic file log

Commit:
Merge pull request #115 from StrongMonkey/add-panic

add panic file log

ibuildthecloud committed:
Don't use /var/lib/docker anymore

ibuildthecloud committed:
Merge pull request #116 from ibuildthecloud/no-var-lib-docker

Don't use /var/lib/docker anymore

Commit:
Disable cgo on build

Commit:
don't use strong type for DockerInspect.State

ibuildthecloud committed:
Merge pull request #120 from StrongMonkey/healthcheck-fix

don't use strong type for DockerInspect.State

Commit:
Add mac address as a label

Commit:
Merge pull request #121 from cjellick/add-mac-label

Add mac address as a label

Commit:
Dont use /dev/sda in tests

Commit:
Add mutex to protect against race condition

Commit:
Merge pull request #123 from cjellick/racefix

Add mutex to protect against race condition

ibuildthecloud committed:
Merge pull request #122 from cjellick/dont-use-dev-sda

Dont use /dev/sda in tests

imikushin committed:
Attach rancher storage volumes to the host before starting container

Commit:
Check state file for CNI IP and error

Commit:
don't lose decimal for flt

ibuildthecloud committed:
Merge pull request #124 from cjellick/cni-state-file

Check state file for CNI IP and error

ibuildthecloud committed:
Merge pull request #119 from cjellick/disable-cgo

Disable cgo on build

Commit:
Merge pull request #128 from StrongMonkey/host-stats-fix

don't lose decimal for float number

Commit:
Revert "don't lose decimal for float number"

Commit:
Merge pull request #129 from rancher/revert-128-host-stats-fix

Revert "don't lose decimal for float number"

Commit:
Close logs when user cancels request

Commit:
Don't use go routine for exec reader

The combination of the go routine and select {} was orphaning the go
routine that contained the select {}.

Commit:
try three times then return success

Commit:
Merge pull request #131 from StrongMonkey/volume-fix

try three times then return success

ibuildthecloud committed:
Merge pull request #130 from cjellick/kill-logs

Close logs when user cancels request

Commit:
small fix

ibuildthecloud committed:
Merge pull request #132 from StrongMonkey/fix-2

small fix

ibuildthecloud committed:
Generate and send host rsa key

Commit:
Merge pull request #118 from StrongMonkey/scheduler

Populate io.rancher.host.ip_set label in each host

ibuildthecloud committed:
Merge pull request #133 from ibuildthecloud/key

Generate and send host rsa key

ibuildthecloud committed:
Merge pull request #127 from imikushin/attach

Attach managed volumes to the host before starting container

Commit:
fix stats bugs

cloudnautique committed:
Switch to dir instead of base

Commit:
Merge pull request #135 from cloudnautique/fix_agent_issues

Switch to dir instead of base

Commit:
fix image pull log

Commit:
bump to 1.7.5

Commit:
fix agent leak

Commit:
Merge pull request #138 from StrongMonkey/agent-leak

fix agent leak

Commit:
don't refresh ip label

Commit:
Merge pull request #140 from StrongMonkey/no-refresh

don't refresh ip label

ibuildthecloud committed:
Add 1.13.1 matrix build

ibuildthecloud committed:
Merge pull request #141 from ibuildthecloud/113

Add 1.13.1 matrix build

Commit:
Merge pull request #136 from StrongMonkey/storage-log

fix image pull log

Commit:
Merge pull request #137 from StrongMonkey/go1.7.5

bump to 1.7.5

Commit:
don't use timeout client in instance.stop

Commit:
Merge pull request #134 from StrongMonkey/stats-fix

fix stats bugs

Commit:
Merge pull request #142 from StrongMonkey/timeout-stop

don't use timeout client in instance.stop

Commit:
add command line support for require_any

Commit:
Merge pull request #143 from StrongMonkey/require_any_label

add command line support for require_any

ibuildthecloud committed:
Use more workers

Commit:
update vendor

Commit:
tweak test

ibuildthecloud committed:
Merge pull request #146 from ibuildthecloud/more-workers

Use more workers

Commit:
Merge pull request #145 from StrongMonkey/update-websocket

fix websocket ping

Commit:
Properly parse IP from cni state file

The CNI plugins report the IP address in cidr notation: 0.0.0.0/32
So, when we are getting the IP address from the file, we need to
properly parse it instead of just using the entire string.

Commit:
Merge pull request #147 from cjellick/ip-read-fix

Properly parse IP from cni state file

Commit:
exec tty resize support

Commit:
fix for externalID and ip label

Commit:
Merge pull request #149 from StrongMonkey/ipfix

fix for externalID and ip label

Commit:
Merge pull request #148 from loganhz/master

exec tty resize support

Commit:
Add hello-world:latest to image pull

Commit:
Merge pull request #150 from cjellick/master

Add hello-world:latest to image pull

Commit:
update vendor

Commit:
don't use timeout client when pulling images

Commit:
add cloud provider

Commit:
send error back if error has retry

Commit:
Merge pull request #152 from StrongMonkey/pull-fix

don't use timeout client when pulling images

ibuildthecloud committed:
Serialize compute calls for devicemapper

ibuildthecloud committed:
Merge pull request #155 from ibuildthecloud/single-file

Serialize compute calls for devicemapper

ibuildthecloud committed:
Switch to smaller test images

ibuildthecloud committed:
Merge pull request #156 from ibuildthecloud/smaller

Switch to smaller test images

ibuildthecloud committed:
Smaller images round 2

ibuildthecloud committed:
Merge pull request #157 from ibuildthecloud/smaller2

Smaller images round 2

ibuildthecloud committed:
Make compute lock be more fine grained for start

On container activate we may need to pull an image or wait for container
IP to be assigned.  During this time we do not want to block other
container operations.

Commit:
Merge pull request #158 from ibuildthecloud/serial2

Make compute lock be more fine grained for start

Commit:
pull image second time w/o credential if first time failed w/ credential

Commit:
remove 1.11 from the matrix

ibuildthecloud committed:
Merge pull request #159 from StrongMonkey/image-fix1

pull image second time w/o credential if first time failed w/ credential

Commit:
Merge pull request #160 from StrongMonkey/remove-1.11

remove 1.11 from the matrix

Commit:
Merge pull request #151 from StrongMonkey/cloud-provider

Populate AWS region and AZ labels

Commit:
Merge pull request #154 from StrongMonkey/retry

send error back if error has retry

Commit:
change label names

Commit:
Merge pull request #164 from StrongMonkey/naming

change label names

Commit:
use instance name in instance.pull

Commit:
Rollback external ID check

This error is not handle properly in service.reconcile.

Commit:
Merge pull request #165 from StrongMonkey/pull

use instance name in instance.pull

Commit:
Merge pull request #166 from cjellick/no-exid-check

Rollback external ID check

ibuildthecloud committed:
Disable hostname verification for proxy to https

Commit:
Merge pull request #167 from ibuildthecloud/disable-verify

Disable hostname verification for proxy to https

imikushin committed:
Add `Init` field

StrongMonkey commented:
will also address https://github.com/rancher/rancher/issues/8498 and https://github.com/rancher/rancher/issues/8643

cjellick commented:
I'd like to see all of our volume based validation tests pass against this PR before merging.
Can you run those tests @StrongMonkey? Start with whatever tests you weant, but rancher-nfs seems like it would be a good starting point

cjellick commented on core/compute/compute_common.go:
```diff
@@ -134,14 +135,27 @@ func getDockerRoot(client *client.Client) string {
 		for _, volume := range volumes {
 			parts := strings.SplitN(volume, ":", 3)
+			// don't set rancher managed volume
+			if utils.SearchInList(rancherManagedVolumeNames, parts[0]) {
```
Could you just use a map instead of a list and check if the key exists or not?

Commit:
update event-subscriber

Commit:
remove bad log

Commit:
runinit

Commit:
Merge pull request #186 from StrongMonkey/initagain

runinit

Commit:
add StopTimeout field

Commit:
fix json unmarshall error for getting conn token

Commit:
Add new CPU-related fields

Commit:
improve logging

Commit:
Merge pull request #207 from StrongMonkey/fix-token

fix json unmarshall error for getting conn token v1.6

Commit:
Unset change name of CPUset

Commit:
switch to 1.8.3

Commit:
get it to compile

aiwantaozi committed:
Add vendor files for Aliyun

aiwantaozi committed:
Add Aliyun Cloud Provider for agent

aiwantaozi committed:
improve cloud provider retry stretagy

aiwantaozi committed:
add flag to control cloud provider

Commit:
Inject DNS client server into windows containers

Commit:
Populate os and kernel information

Commit:
Cross-compile

And fix go fmt error

Commit:
fix golint error

Commit:
Support RANCHER_DEBUG

aiwantaozi committed:
add dns resolv check

Commit:
new flex volume changes

Commit:
Trim key and value for user entered labels

Get rid of trailing and leading whitespace in user provided labels.
This is the best way to ensure that the label keys and values have
consistent values throughout Rancher.

Commit:
Merge pull request #251 from cjellick/fix-fix-label-fix

Trim key and value for user entered labels

Commit:
Windows agent start-up script

Add a new script for running agent in windows.
This script can support adding labels and setting agent ip.

Commit:
Add port mapping support for windows transparent network

Use per-host-subnet to support port mapping network.
Remove the port configuration when creating instance.

Commit:
Merge pull request #252 from orangedeng/my-windows

Rancher windows enhancement

Commit:
Update aliyun SDK to fix metadata issue

rancher/rancher#10439
rancher/rancher#10474

Commit:
Fix that CI does not package windows binary

When we building release binaries, 'CROSS=1' will be set to env
var. Fix conditional of $CROSS in package script.

Commit:
Change the label name of skip host healthcheck.

Use `io.rancher.host.healthcheck.deploy_strategy` instead of
    `io.rancher.infra_service.healthcheck.deploy` for consistency.
More extensiablity as using strategy subfix.

Commit:
Update version of github.com/microsoft/go-winio in trash.yml

Commit:
Update vendors of github.com/Microsoft/go-winio

refer to https://github.com/rancher/rancher/issues/10536
We need to update version of github.com/Microsoft/go-winio
as we using golang 1.8.3.

Commit:
Merge pull request #258 from orangedeng/windows-fix

Update dependence version of microsoft/go-winio

Commit:
Use Agent IP Env Var if Exists for Windows Agent

refer to https://github.com/rancher/rancher/issues/10590

Commit:
Set DNS Search List in Windows Containers

refer to bug https://github.com/rancher/rancher/issues/10578

Commit:
Merge pull request #259 from orangedeng/windows-fix

Bug fix for agent in windows environment

Commit:
Change DNS sequence for windows container

refer to https://github.com/rancher/rancher/issues/10626
Configure 169.254.169.251 as the first dns address in Windows container.
